src/rust/engine/src/context.rs

```diff
@@ -212,63 +213,64 @@ impl Core {
     exec_strategy_opts: &ExecutionStrategyOptions,
     remoting_opts: &RemotingOptions,
   ) -> Result<Box<dyn CommandRunner>, String> {
-    if (exec_strategy_opts.remote_cache_read || exec_strategy_opts.remote_cache_write)
-      && remoting_opts.execution_enable
-    {
-      return Err(
-        "Remote caching mode and remote execution mode cannot be enabled concurrently".into(),
-      );
-    }
+    let remote_caching_used =
+      exec_strategy_opts.remote_cache_read || exec_strategy_opts.remote_cache_write;
+
+    // If remote caching is used with eager validation, we do not want to use the remote store
+    // with the local command runner. This reduces the surface area of where the remote store is
+    // used to only be the remote cache command runner.
+    let local_only_store = if remote_caching_used && remoting_opts.store_eager_fetch {
```

Comment:
Since this will sometimes be local only, and sometimes not, you might call it something else. `store_maybe_local_only` is a mouthful, but...